be/src/runtime/local_pass_through_buffer.cpp

```diff
@@ -4,33 +4,50 @@
 
 #include "column/chunk.h"
 #include "common/logging.h"
+#include "runtime/current_thread.h"
+#include "runtime/mem_tracker.h"
 
 namespace starrocks {
 
 // channel per [sender_id]
 class PassThroughSenderChannel {
 public:
-    void append_chunk(const vectorized::Chunk* chunk, size_t chunk_size, int32_t driver_sequence) {
-        auto clone = chunk->clone_unique();
-        {
-            std::unique_lock lock(_mutex);
-            _buffer.emplace_back(std::make_pair(std::move(clone), driver_sequence));
-            _bytes.push_back(chunk_size);
+    PassThroughSenderChannel() = default;
+
+    ~PassThroughSenderChannel() {
+        if (_physical_bytes > 0) {
+            CurrentThread::current().mem_consume(_physical_bytes);
         }
     }
 
+    void append_chunk(const vectorized::Chunk* chunk, size_t chunk_size, int32_t driver_sequence) {
+        // Release allocated bytes in current MemTracker, since it would not be released at current MemTracker
+        int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
+        auto clone = chunk->clone_unique();
+        int64_t physical_bytes = CurrentThread::current().get_consumed_bytes() - before_bytes;
+        DCHECK_GE(physical_bytes, 0);
+        CurrentThread::current().mem_release(physical_bytes);
+
+        std::unique_lock lock(_mutex);
+        _buffer.emplace_back(std::make_pair(std::move(clone), driver_sequence));
+        _bytes.push_back(chunk_size);
+        _physical_bytes += physical_bytes;
+    }
     void pull_chunks(ChunkUniquePtrVector* chunks, std::vector<size_t>* bytes) {
-        {
-            std::unique_lock lock(_mutex);
-            chunks->swap(_buffer);
-            bytes->swap(_bytes);
-        }
+        std::unique_lock lock(_mutex);
+        chunks->swap(_buffer);
+        bytes->swap(_bytes);
+
+        // Consume physical bytes in current MemTracker, since later it would be released
+        CurrentThread::current().mem_consume(_physical_bytes);
+        _physical_bytes = 0;
     }
 
 private:
     std::mutex _mutex; // lock-step to push/pull chunks
     ChunkUniquePtrVector _buffer;
     std::vector<size_t> _bytes;
+    int64_t _physical_bytes = 0; // Physical consumed bytes for each chunk
```

Comment:
Is it possiable cause a race condition here?

Reply:
All of them are protected with the `_mutex`.